vault/expiration.go

```diff
@@ -505,6 +505,28 @@ func (m *ExpirationManager) revokeCommon(leaseID string, force, skipToken bool)
 		return nil
 	}
 
+	// If not in force mode, where we want to ignore errors, and not in
+	// skiptoken mode where we're already in the middle of a revocation
+	// operation, if the expiration time has not already been hit, set
+	// expiration to now, persist, and call updatePending to hand off
+	// revocation to the expiration manager's pending timer map. We still
+	// attempt to continue revocation here.
+	if !force && !skipToken && le.ExpireTime.After(time.Now()) {
+		le.ExpireTime = time.Now()
+		{
+			m.pendingLock.Lock()
+			if err := m.persistEntry(le); err != nil {
+				m.pendingLock.Unlock()
+				return err
+			}
+
+			m.updatePendingInternal(le, 0)
+			m.pendingLock.Unlock()
+
+			return nil
```

Comment:
Why return here? And if intended, then I'm not following the comment, "We still attempt to continue revocation here."

Reply:
Forcing the expiration time to zero will immediately trigger an expiration based revocation. In practical terms that means that this function will be called again and potentially simultaneously since it'd be in a different goroutine. So returning here ensures only one attempt actually happens at a time.